confdb/aclocal_util.m4

```diff
@@ -64,7 +64,7 @@ dnl argument is already present in the variable
 AC_DEFUN([PAC_APPEND_FLAG],[
 	AC_REQUIRE([AC_PROG_FGREP])
 	AS_IF(
-		[echo "$$2" | $FGREP -e "\<$1\>" >/dev/null 2>&1],
+		[echo "$$2" | $FGREP -e "$1" >/dev/null 2>&1],
```

Comment:
I guess I was assuming some "basic regular expression" support when I added those in https://github.com/pmodels/mpich/commit/f2d0f7fe87a42e2b751e77a203b157249ed2d1f6. It seemed to work on the platforms I tested on 🤷. https://pubs.opengroup.org/onlinepubs/9699919799/utilities/grep.html

Reply:
```
-F
    Match using fixed strings. Treat each pattern specified as a string instead of a regular expression. If an input line contains any of the patterns as a contiguous sequence of bytes, the line shall be matched. A null string shall match every line.
```

We could replace `$FGREP` with `$GREP` (and with `AC_PROG_GREP`). `$FGREP` prevents special chars in `$1` and should be faster than `$GREP`.

With `grep`, we could use pattern such as `"\(^\| \)$1\($\| \)"`.

Since it is imperfect either way, I tend to keep it on the simple side (fix the local breakage locally).

Reply:
Hm, now I see. I guess this never worked 😦.

Reply:
Also note, that "\<$1\>" does not work even with `grep` or `egrep` when `$1` contains non-word characters.